runtime/etc/Makefile.mli-static

```diff
@@ -0,0 +1,81 @@
+# Copyright 2004-2019 Cray Inc.
+# Other additional copyright holders may be indicated within.
+#
+# The entirety of this work is licensed under the Apache License,
+# Version 2.0 (the "License"); you may not use this file except
+# in compliance with the License.
+#
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+#
+# Makefile.mli-static
+#
+
+#
+# TODO
+#
+# There's a slightly more complicated build process going on here, because we
+# have 3 deliverables (for now):
+#
+#   - A client facing library file (static or shared)
+#   - A multi-locale launcher
+#   - A multi-locale executable
+#
+# We build the server as if it's a multi-locale executable, and then build the
+# client as if it's a (static, for this file) library.
+#
+
+include $(CHPL_MAKE_HOME)/runtime/etc/Makefile.include
+
+#
+# This is common code included wholesale in each MLI compile (for now).
+#
+CHPL_MLI_INCLUDES = $(CHPL_MAKE_HOME)/runtime/etc/mli
+
+#
+# You MUST SET THIS or compilation will fail!
+#
+ifndef CHPL_ZMQ_LOC
+	$(error CHPL_ZMQ_LOC must be set to the location of the ZeroMQ library)
+endif
+
+CHPL_LINK_TRAILING = $(CHPL_MAKE_THIRD_PARTY_LINK_ARGS) $(CHPL_MAKE_BASE_LFLAGS)
+
+all: $(TMPCLIENTNAME) $(TMPSERVERNAME)
+
+# TODO: Should I roll a static copy of libzmq into the client archive?
+$(TMPCLIENTNAME): $(CHPL_CL_OBJS) FORCE
+	$(CC) $(CHPL_MAKE_BASE_CFLAGS) $(GEN_CFLAGS) $(COMP_GEN_CLIENT_CFLAGS) -c -o $(TMPCLIENTNAME).o $(CHPL_RT_INC_DIR) $(CHPL_CLIENT_SRC) -I$(CHPL_MLI_INCLUDES) -I$(CHPL_ZMQ_LOC)/include
+	$(AR) -c -r -s $(TMPCLIENTNAME) $(TMPCLIENTNAME).o $(CHPL_CL_OBJS) $(CHPL_ZMQ_LOC)/lib/libzmq.a
+ifneq ($(TMPCLIENTNAME),$(CLIENTNAME))
+	cp $(TMPCLIENTNAME) $(CLIENTNAME)
+	rm $(TMPCLIENTNAME)
+endif
+
+# Should I add the $(CHPL_CL_OBJS) dependency twice?
```

Comment:
Why?

Reply:
Hmmmm, not sure. I can't recall why I wrote that.